airbyte-ci/connectors/metadata_service/lib/tests/test_gcs_upload.py

```diff
@@ -266,107 +376,102 @@ def test_upload_metadata_to_gcs_valid_metadata(
         expected_version_doc_key = f"metadata/{metadata.data.dockerRepository}/{metadata.data.dockerImageTag}/{DOC_FILE_NAME}"
         expected_latest_doc_key = f"metadata/{metadata.data.dockerRepository}/latest/{DOC_FILE_NAME}"
 
-        latest_blob_exists = latest_blob_md5_hash is not None
-        version_blob_exists = version_blob_md5_hash is not None
-        doc_version_blob_exists = doc_version_blob_md5_hash is not None
-        doc_latest_blob_exists = doc_latest_blob_md5_hash is not None
+        is_release_candidate = getattr(metadata.data.releases, "isReleaseCandidate", False)
 
         # Call function under tests
 
         upload_info = gcs_upload.upload_metadata_to_gcs(
             "my_bucket", metadata_file_path, validator_opts=ValidatorOptions(docs_path=DOCS_PATH)
         )
 
-        # Assertions
+        # Assert correct file upload attempts were made
 
-        assert gcs_upload._metadata_upload.call_count == 2
-        is_release_candidate = getattr(metadata.data.releases, "isReleaseCandidate", False)
-        if is_release_candidate:
-            gcs_upload._metadata_upload.assert_has_calls(
-                [
-                    mocker.call(metadata, mocks["mock_bucket"], metadata_file_path, metadata.data.dockerImageTag),
-                    mocker.call(metadata, mocks["mock_bucket"], metadata_file_path, RELEASE_CANDIDATE_GCS_FOLDER_NAME),
-                ]
-            )
-        else:
-            gcs_upload._metadata_upload.assert_has_calls(
-                [
-                    mocker.call(metadata, mocks["mock_bucket"], metadata_file_path, metadata.data.dockerImageTag),
-                    mocker.call(metadata, mocks["mock_bucket"], metadata_file_path, LATEST_GCS_FOLDER_NAME),
-                ]
-            )
-        gcs_upload._doc_upload.assert_called()
+        expected_calls = [
+            # Always upload the versioned metadata
+            mocker.call(
+                local_file_path=metadata_file_path, bucket=mocks["mock_bucket"], blob_path=expected_version_key, disable_cache=True
+            ),
+            # Always upload the versioned doc
+            mocker.call(
+                local_file_path=VALID_DOC_FILE_PATH,
+                bucket=mocks["mock_bucket"],
+                blob_path=expected_version_doc_key,
+                disable_cache=False,
+            ),
+        ]
 
-        gcs_upload.service_account.Credentials.from_service_account_info.assert_called_with(json.loads(mocks["service_account_json"]))
-        mocks["mock_storage_client"].bucket.assert_called_with("my_bucket")
         if is_release_candidate:
```

Comment:
I wonder if its worth pulling this case out like the prerelease one, so we can nail down the exact behavior we want in this case (see: other comment about release candidate uploads)